huaweicloud/services/acceptance/acceptance.go

```diff
@@ -45,6 +51,184 @@ func init() {
+	var resName, keyName string
+	// Check the format of the variable.
+	match, _ := regexp.MatchString(checkAttrRegexpStr, varStr)
+	if !match {
+		return resName, keyName, fmtp.Errorf("The type of 'variable' is error, "+
+			"expected ${resourceType.name.field} got %s", varStr)
+	}
+
+	reg, err := regexp.Compile(checkAttrRegexpStr)
+	if err != nil {
+		return resName, keyName, fmtp.Errorf("The acceptance function is wrong.")
+	}
+	mArr := reg.FindStringSubmatch(varStr)
+	if len(mArr) < 1 {
+		return resName, keyName, fmtp.Errorf("The type of 'variable' is error, "+
+			"expected ${resourceType.name.field} got %s", varStr)
+	}
+
+	// Get resName and keyName from variable.
+	strs := strings.Split(mArr[1], ".")
+	for i, s := range strs {
+		if strings.Contains(s, "huaweicloud_") {
+			resName = strings.Join(strs[0:i+2], ".")
+			keyName = strings.Join(strs[i+2:], ".")
+			break
+		}
+	}
+	return resName, keyName, nil
+}
+
+/*
+TestCheckResourceAttrWithVariable validates the variable in state for the given name/key combination.
+  Parameters:
+    resourceName: The resource name is used to check in the terraform.State.
+    key:          The field name of the resource.
+    variable:     The variable name of the value to be checked.
+
+    variable such like ${huaweicloud_waf_certificate.certificate_1.id}
+    or ${data.huaweicloud_waf_policies.policies_2.policies.0.id}
+*/
+func TestCheckResourceAttrWithVariable(resourceName, key, varStr string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		resName, keyName, err := parseVariableToName(varStr)
+		if err != nil {
+			return err
+		}
+
+		if strings.EqualFold(resourceName, resName) {
+			return fmtp.Errorf("Meaningless verification. " +
+				"The referenced resource cannot be the current resource.")
+		}
+
+		// Get the value based on resName and keyName from the state.
+		rs, ok := s.RootModule().Resources[resName]
+		if !ok {
+			return fmtp.Errorf("Can't find %s in state : %s.", resName, ok)
+		}
+		value := rs.Primary.Attributes[keyName]
+
+		return resource.TestCheckResourceAttr(resourceName, key, value)(s)
+	}
+}
+
+// CheckResourceDestroy check whether resources destroied in HuaweiCloud.
+func (rc *resourceCheck) CheckResourceDestroy() resource.TestCheckFunc {
+	if strings.Compare(rc.resourceType, dataSourceTypeCode) == 0 {
+		fmtp.Errorf("Error, you built a resourceCheck with 'InitDataSourceCheck', " +
+			"it cannot run CheckResourceDestroy().")
+		return nil
+	}
+	return func(s *terraform.State) error {
+		strs := strings.Split(rc.resourceName, ".")
+		var resourceType string
+		for _, str := range strs {
+			if strings.Contains(str, "huaweicloud_") {
+				resourceType = strings.Trim(str, " ")
+				break
+			}
+		}
+
+		for _, rs := range s.RootModule().Resources {
+			if rs.Type != resourceType {
+				continue
+			}
+
+			conf := TestAccProvider.Meta().(*config.Config)
+			if _, err := rc.getResourceFunc(conf, rs); err == nil {
```

Comment:
should check `getResourceFunc` is nil or not?

Reply:
added